code/modules/mob/living/carbon/human/ai/brain/ai_brain_cover.dm

```diff
@@ -19,6 +19,9 @@
 	if(!COOLDOWN_FINISHED(src, cover_search_cooldown))
 		return
 
+	if(!cover_without_gun && !primary_weapon)
```

Comment:
for some reason this feels off to me but i can't put my finger on it, so here's what all the logic does, make sure it's what you intended
```
!TRUE AND !TRUE - We can cover without guns and we have a gun, allows us to find cover
!TRUE AND !FALSE - We can cover without guns and we don't have a gun, allows us to find cover
!FALSE AND !TRUE - We can't cover without guns and we have a gun, allows us to find cover
!FALSE AND !FALSE - We can't cover without guns and we don't have a gun, stops us from finding cover
```

Reply:
Somehow this seemed alright ingame, but now i can say that the logic is indeed faulty